### DISCUSSION TITLE
Opening links in new tab/window

### DISCUSSION THREAD

**Discussion by accforgithubtest** [sqlpage/SQLPage#235] [Q&A]
Hello fellow sqlpagers,

I have some data that links to external sites, which is rendered in a table component, with `columnName as markdown` enabled for the table/column in component. I am looking for a way to open the links in a new tab/window (the current default is they are opening in sqlpage window/tab itself). 

Normally this would be done via a target="_blank" in html, and based on my googling, it looks like markdown itself doesn't have a standard way to specify  target="_blank".

Is there any way to render links in sqlpage and force it to open in a new tab/window ? 

TIA for your suggestions and help !

**Comment by lovasoa** [comment]
Hello ! 

  This is currently not possible, because the markdown parser that we use [is not configurable enough](https://github.com/wooorm/markdown-rs/issues/32). We could handle this by adding some new attributes to the table component, though.

  Anyway, I agree we should have more options to render more complicated things inside tables. I'll open a new feature request.

**Comment by lovasoa** [comment]
I opened https://github.com/lovasoa/SQLpage/issues/236

**Comment by olivierauverlot** [comment]
Hello,

The solution could be create a custom table component with an advanced support of links. 

They are so possibilities with table in web applications. That seems to me difficult to provide a generic component that satisfy all needs.

**Reply by lovasoa** [reply]
What I'd love to get to is to get the ability to embed one component into another one. We are still far away from that, but ideally one would be able to do something like

```sql
select 'table' as component, 'my_column' as embed;
select '[
    {"component": "button", "new_tab": true},
    {"link": "http://...", "title": "Click me"}
]' as my_column; 
```

and even, after we do both https://github.com/lovasoa/SQLpage/issues/20 and https://github.com/lovasoa/SQLpage/issues/237

```sql
select 'table' as component, 'my_column' as embed;
select sqlpage.run_sql('my_complex_layout.sql?id=' || id) as my_column from my_table; 
```
